File: solutions/2952. Minimum Number of Coins to be Added/2952.java
class Solution {
  // Same as 330. Patching Array
  public int minimumAddedCoins(int[] coins, int target) {
    int ans = 0;
    int i = 0;     // coins' index
    long miss = 1; // the minimum sum in [1, n] we might miss

    Arrays.sort(coins);

    while (miss <= target)
      if (i < coins.length && coins[i] <= miss) {
        miss += coins[i++];
      } else {
        // Greedily add `miss` itself to increase the range from
        // [1, miss) to [1, 2 * miss).
        miss += miss;
        ++ans;
      }

    return ans;
  }
}

File: solutions/2952. Minimum Number of Coins to be Added/2952.py
class Solution:
  # Same as 330. Patching Array
  def minimumAddedCoins(self, coins: List[int], target: int) -> int:
    ans = 0
    i = 0  # coins' index
    miss = 1  # the minimum sum in [1, n] we might miss

    coins.sort()

    while miss <= target:
      if i < len(coins) and coins[i] <= miss:
        miss += coins[i]
        i += 1
      else:
        # Greedily add `miss` itself to increase the range from
        # [1, miss) to [1, 2 * miss).
        miss += miss
        ans += 1

    return ans
